contrib/coredns/policy/policy.go

```diff
@@ -119,6 +119,52 @@ func (p *PolicyMiddleware) getEDNS0Attrs(r *dns.Msg) ([]*pb.Attribute, bool) {
 	return attrs, foundSourceIP
 }
 
+
+type LocalResponseWriter struct {
+	dns.ResponseWriter
```

Comment:
I don't think we actually need to keep the response writer in here. All this LocalResponseWriter is trying to do is capture any response locally - not to write it back to the client.